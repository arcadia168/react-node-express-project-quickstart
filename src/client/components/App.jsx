import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from "react-router-dom";
import { Button } from 'react-bootstrap';
import PropTypes from 'prop-types'

const HelloWorld = () => <h1 data-testid="app-title" className="app__title">Hello World</h1>;

const App = () => {
  return (
    <Router>
      <Route exact path="/" component={HelloWorld} />
      <Button>Hello React Start with Bootstrap!</Button>
    </Router>
  )
}

App.propTypes = {
  axiosInstance: PropTypes.func,
}

export default App;
