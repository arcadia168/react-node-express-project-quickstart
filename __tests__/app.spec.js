import React from 'react';
import App from '../src/client/components/App';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe('App', () => {
    const render = () => {
        return mount(<App/>);
    }

    it('renders the app as expected', () => {
        const component = renderer.create(
            <App />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('successfully renders the app', () => {
        const renderedApp = render();
        expect(
            renderedApp.find('[data-testid="app-title"]').text(),
        ).toBe('Hello World');
    });
});

