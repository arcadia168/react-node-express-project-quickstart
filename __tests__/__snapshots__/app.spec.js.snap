// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the app as expected 1`] = `
<h1
  data-testid="app-title"
>
  Hello World
</h1>
`;

exports[`Link changes the class when hovered 1`] = `
<h1>
  Hello World
</h1>
`;
